### PR TITLE
Dont allow taxonomies without show ui to be excluded

### LINKED ISSUE #439: Prominent Keywords appear as Terms to Exclude
### What did you expect to happen?

Prominent keyword would _not_ appear as a term to exclude. Note, that categories still show-up. 

### What happened instead?

Prominent Keywords appear as Terms to Exclude.

### How can we reproduce this behavior?

**1.Add News SEO**

**2.Create a post with some content.**

 **3.Select Posts (or Pages) and see prominent keywords appear as terms to exclude.**

screen_shot_2018-10-11_at_8_36_59_am

Can you provide a link to a page which shows this issue?

### Technical info
* WordPress version: 4.9.8
* Yoast SEO version: Yoast SEO Premium 9 RC 1
* News SEO 9 RC1. Not a regression.

### PR DESCRIPTION
**Pull request by Xyfi**
## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where prominent words would be shown in the list of terms to be excluded from the news sitemap.

## Relevant technical choices:

* Create a class `WPSEO_Exclude_Taxonomies` with a static helper function to retrieve all Taxonomies of which the terms are eligible for exclusion. 
* After consulting with @jip and @jdevalk we decided to exclude taxonomies from the list of with the `show_ui` parameter is false.

## Test instructions

This PR can be tested by following these steps:

* Make sure you use Yoast SEO Premium.
* Before you check out this branch check whether you have prominent words in the News SEO admin page, under `Terms to exclude for Posts`.
* If not create a few posts with similar content, and check again.
* Check out this branch, make sure the prominent words no longer apprear.

Fixes #439

### CODE REVIEW

File: classes/class-exclude-taxonomies.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * Class containing utility functions for excluding posts with certain terms from the news sitemap.
+ */
+class WPSEO_Exclude_Taxonomies {
```
Comment (andizer) [inline classes/class-exclude-taxonomies.php]: Please use `WPSEO_News_` as prefix.

File: classes/class-exclude-taxonomies.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	/**
+	 * Gets a list of taxonomies of which posts with terms of this taxonomy can be excluded from the sitemap.
+	 *
+	 * @param string $post_type The post type.
+	 *
+	 * @return array Taxonomies of which posts with terms of this taxonomy can be excluded from the sitemap.
+	 */
+	public static function getExcludableTaxonomies( $post_type ) {
```
Comment (andizer) [inline classes/class-exclude-taxonomies.php]: get_exclude_taxonomies

File: classes/class-admin-page.php [hunk 3]
```diff
@@ -146,8 +146,10 @@ protected function filter_included_post_type( $post_type ) {
 	private function get_excluded_post_type_taxonomies( $post_type ) {
 		$terms_per_taxonomy = array();
 
-		foreach ( get_object_taxonomies( $post_type->name, 'objects' ) as $taxonomy ) {
-			$terms = get_terms( array( 'taxonomy' => $taxonomy->name, 'hide_empty' => false ) );
+		$taxonomies = WPSEO_Exclude_Taxonomies::getExcludableTaxonomies( $post_type->name );
```
Comment (andizer) [inline classes/class-admin-page.php]: This is the only location where it  has been used. I would choose for a non static approach.

File: classes/class-admin-page.php [hunk 4]
```diff
@@ -146,8 +146,10 @@ protected function filter_included_post_type( $post_type ) {
...
+
+		foreach ( $taxonomies as $taxonomy ) {
+			$terms = get_terms( array( 'taxonomy' => $taxonomy->name, 'hide_empty' => false, 'show_ui' => true ) );
```
Comment (andizer) [inline classes/class-admin-page.php]: I would wrap this in a method that is in the `WPSEO_Exclude_Taxonomies ` class.

File: classes/class-admin-page.php [hunk 5]
```diff
@@ -146,8 +146,10 @@ protected function filter_included_post_type( $post_type ) {
 	private function get_excluded_post_type_taxonomies( $post_type ) {
```
Comment (andizer) [inline classes/class-admin-page.php]: You can move the whole content of this method to the `excluded_taxomies` class.

**Comment by Dieterrr** [comment]
CR done 👍

**Comment by jcomack** [comment]
Acceptance done 👍

**Pull request merged**